### PR TITLE
[MM-57194] Mark confidential settings as secrets

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Make use of https://github.com/mattermost/mattermost/pull/27986 by marking the following config settings as secrets:
- `GitHubOAuthClientSecret`
- `WebhookSecret`
- `EncryptionKey`

The other changes are needed due to the breaking changes in https://github.com/mattermost/mattermost/pull/26471.

`GitHubOAuthClientID` is shown in plain text in the GitHub UI, so I don't think it needs to be obfuscated.

Screenshot from 2024-08-15 13-09-18

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57194

### CODE REVIEW

File: server/plugin/flows.go [hunk 1]
```diff
@@ -79,31 +83,49 @@ func (p *Plugin) NewFlowManager() *FlowManager {
...
 		fm.stepOAuthInfo(),
 		fm.stepOAuthInput(),
 		fm.stepOAuthConnect().Terminal(),
 
 		fm.stepCancel("setup oauth"),
 	)
-	fm.webhokFlow = fm.newFlow("webhook").WithSteps(
+	fm.oauthFlow = oauthFlow
+
+	webhokFlow, err := fm.newFlow("webhook")
```
Comment (wiggin77) [inline server/plugin/flows.go:102]: Is this meant to be 'webHookFlow'?

Comment (hanzei) [reply server/plugin/flows.go]: Good catch, fixe the typo

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Comment by hanzei** [comment]
https://github.com/mattermost/mattermost/pull/27986#issuecomment-2339638119

**Review by AayushChaudhary0001 — APPROVED** [review]
This PR has been tested with the help of https://github.com/mattermost/mattermost/pull/27986 as it requires some changes in the server.
All the secrets and keys are shown encrypted in the plugin settings, LGTM. Approved.

**Pull request merged**